### DISCUSSION TITLE
Device = cuda ———> in Colab not working

### DISCUSSION THREAD

**Discussion by Alisoltan82** [Project-MONAI/tutorials#1606] [Q&A] [answered]
While attempting to run with colab gpu , while torch.cuda is getting the device name , the whole body model deliver error of no cuda detected and accept the cpu run!!!

Kindly advice on cuda version at colab or the inferer file solution where I tried everything with the inference file with no result

**Comment by KumoLiu** [comment]
Hi @Alisoltan82, which tutorial did you refer to?

**Reply by Alisoltan82** [reply]
Hi @KumoLiu 

https://github.com/lukepolson/youtube_channel/blob/3642cdd80f9200a5db4e622a3fe2c1a8f6868ecd/Python%20Metaphysics%20Series/vid40.ipynb


I'm trying to run on a gpu and changed the device to cuda

here is my model
```
device = torch.device('cuda:0')
torch.cuda.get_device_name()
 'Tesla P100-PCIE-16GB'

model = config.get_parsed_content("network").to(device)
model.load_state_dict(torch.load(model_path))

```

and this is the error message when run

```
NotImplementedError: Could not run 'aten::slow_conv3d_forward' with arguments from the 'CUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::slow_conv3d_forward' is only available for these backends: [CPU, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradHIP, AutogradXLA, AutogradMPS, AutogradIPU, AutogradXPU, AutogradHPU, AutogradVE, AutogradLazy, AutogradMeta, AutogradMTIA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, AutogradNestedTensor, Tracer, AutocastCPU, AutocastCUDA, FuncTorchBatched, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PythonDispatcher].

```

**Comment by ericspod** [comment, marked as answer]
This may be because the weights were stored with a certain device that isn't the exact one you're using. Can you try:

```python
model = config.get_parsed_content("network").to("cpu")
model.load_state_dict(torch.load(model_path, map_location="cpu"))
model=model.to(device)
```

**Reply by Alisoltan82** [reply]
Thanks @ericspod  and @KumoLiu 
solved
actually  I'm relatively new to pytorch but I confirmed the device is cuda and added .to(device) in the inference line which started the gpu
```
data['pred'] = inferer(data['image'].unsqueeze(0).to(device), network=model)
```

**Reply by ericspod** [reply]
Yes everything does need to be on the same device like this. Even though the weights were stored on a particular device (instead of cpu) the error you get should have been more about devices. It's an odd Pytorch thing I guess.
